### PR TITLE
Closes #1848: Fix PHPstan errors since updating to phpstan-drupal 1.1.25.

### LINKED ISSUE #1848: PHPstan errors after upgrading to phpstan-drupal to 1.1.25
## Problem/Motivation
Running `phpstan analyse` is failing with errors since we merged az-digital/az-quickstart-dev#50.
 
### Describe the bug
```
$ /src/az-quickstart-scaffolding/vendor/bin/phpstan analyse --configuration web/profiles/custom/az_quickstart/phpstan.neon web/profiles/custom/az_quickstart

 ------ ---------------------------------------------------------------------- 
  Line   modules/custom/az_card/src/Plugin/Field/FieldWidget/AZCardWidget.php  
 ------ ---------------------------------------------------------------------- 
  123    Call to an undefined method                                           
         Drupal\Core\TypedData\TypedDataInterface::isEmpty().                  
  166    Access to an undefined property                                       
         Drupal\Core\TypedData\TypedDataInterface::$link_title.                
  166    Access to an undefined property                                       
         Drupal\Core\TypedData\TypedDataInterface::$link_uri.                  
  167    Access to an undefined property                                       
         Drupal\Core\TypedData\TypedDataInterface::$link_uri.                  
  263    Call to an undefined method                                           
         Drupal\Core\TypedData\TypedDataInterface::isEmpty().                  
 ------ ---------------------------------------------------------------------- 

 [ERROR] Found 5 errors                                                         

$ exit
```

### To Reproduce
Run `phpstan analse` (`lando phpstan`)

## Proposed resolution
Add a variable type hint to `AZCardWidget` to help PHPstan out.

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Should resolve these PHPstan errors since we merged az-digital/az-quickstart-dev#50:
```
$ /src/az-quickstart-scaffolding/vendor/bin/phpstan analyse --configuration web/profiles/custom/az_quickstart/phpstan.neon web/profiles/custom/az_quickstart

 ------ ---------------------------------------------------------------------- 
  Line   modules/custom/az_card/src/Plugin/Field/FieldWidget/AZCardWidget.php  
 ------ ---------------------------------------------------------------------- 
  123    Call to an undefined method                                           
         Drupal\Core\TypedData\TypedDataInterface::isEmpty().                  
  166    Access to an undefined property                                       
         Drupal\Core\TypedData\TypedDataInterface::$link_title.                
  166    Access to an undefined property                                       
         Drupal\Core\TypedData\TypedDataInterface::$link_uri.                  
  167    Access to an undefined property                                       
         Drupal\Core\TypedData\TypedDataInterface::$link_uri.                  
  263    Call to an undefined method                                           
         Drupal\Core\TypedData\TypedDataInterface::isEmpty().                  
 ------ ---------------------------------------------------------------------- 

 [ERROR] Found 5 errors                                                         

$ exit
```
## Related issues
Closes #1848

## How to test

`phpstan analyse` (`lando phpstan`) or look at results in ProboCI.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by trackleft — APPROVED** [review]
https://f6eab5a3-35c5-4de2-a37c-1ca8666fa421.probo.build/pages/card-decks still renders correctly.		
Card preview renders correctly
Existing card values exist correctly 
Demo content still works.
New cards still work as expected.

File: modules/custom/az_card/src/Plugin/Field/FieldWidget/AZCardWidget.php [hunk 1]
```diff
@@ -105,6 +105,9 @@ public function form(FieldItemListInterface $items, array &$form, FormStateInter
    */
   public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
 
+    /** @var \Drupal\az_card\Plugin\Field\FieldType\AZCardItem $item */
```
Comment (trackleft) [inline modules/custom/az_card/src/Plugin/Field/FieldWidget/AZCardWidget.php:108]: Oh, now I see the delta still being used. Nice.

**Review by bberndt-uaz — APPROVED** [review]
Tested in Probo: cards working as expected.

**Comment by tadean** [comment]
Seems to work. You win this round, phpstan...

**Review by tadean — APPROVED** [review]

**Pull request merged**